src/AcmeClient.php

```diff
@@ -146,87 +141,84 @@ public function get(string $resource): Promise
+            ]);
+
+            try {
+                if ($request->getMethod() === 'POST') {
+                    $request->setHeader('content-type', 'application/jose+json');
+                }
+
+                $response = $this->httpClient->request($request);
+                $statusCode = $response->getStatus();
+                $body = $response->getBody()->buffer();
+                $response->setBody(new ReadableBuffer($body));
+
+                $this->logger->debug('Request for {url} via POST has been processed with status {status}: {body}', [
                     'url' => $url,
-                    'body' => $requestBody,
+                    'status' => $statusCode,
+                    'body' => $body,
                 ]);
 
-                try {
-                    if ($request->getMethod() === 'POST') {
-                        $request->setHeader('content-type', 'application/jose+json');
-                    }
+                $this->saveNonce($response);
 
-                    /** @var Response $response */
-                    $response = yield $this->httpClient->request($request);
-                    $statusCode = $response->getStatus();
-                    $body = yield $response->getBody()->buffer();
-
-                    $this->logger->debug('Request for {url} via POST has been processed with status {status}: {body}', [
-                        'url' => $url,
-                        'status' => $statusCode,
-                        'body' => $body,
-                    ]);
-
-                    $this->saveNonce($response);
-
-                    if ($statusCode === 400) {
-                        $info = \json_decode($body, true, 16, \JSON_THROW_ON_ERROR);
-
-                        if (!empty($info['type']) && (\strpos($info['type'], "acme:error:badNonce") !== false)) {
-                            $this->nonces = [];
-                            continue;
-                        }
-                    } elseif ($statusCode === 429) {
-                        /**
-                         * Hit rate limit.
-                         * @{link} https://letsencrypt.org/docs/rate-limits/
-                         */
-                        yield delay(1000);
+                if ($statusCode === 400) {
+                    $info = \json_decode($body, true, 16, \JSON_THROW_ON_ERROR);
+
+                    if (!empty($info['type']) && (\strpos($info['type'], "acme:error:badNonce") !== false)) {
+                        $this->nonces = [];
                         continue;
                     }
-                } catch (Throwable $e) {
-                    throw new AcmeException("POST request to {$url} failed: " . $e->getMessage(), null, $e);
+                } elseif ($statusCode === 429) {
+                    /**
+                     * Hit rate limit.
+                     * @{link} https://letsencrypt.org/docs/rate-limits/
+                     */
+                    delay(1000);
```

Comment:
```suggestion
                    delay(1);
```

Time unit is seconds now.